### PR TITLE
Use JSON for class data

### LINKED ISSUE #25: Convert `class_data` to dict and store as JSON
Currently format for .cld class data file is:
```
f'{student_name}, {avatar_filename}\n' for every student in the class. 
```
Which is bad because it has to be unnecessarily processed when reloaded. JSON would seem to be a much better and less error-prone solution (open to advice).
Saving as a JSON'd dict means we can directly de-JSON the dict and use it. 
Pass that dict to the score-entry logic, add the scores (or `None`) for each student, pass to the image generation. 
Nice and simple. 

Using the student name as a key, the uncleaned input/real name, and the filename of the student's avatar:
```
class_data = { 
    'student_name_1': ['student 1 real name', 'student1_avatar_file_name'],
    'student_name_2': ['student 2 real name', 'student2_avatar_file_name'],
    'student_name_3': ['student 3 real name', 'student3_avatar_file_name'],
    'student_name_4': ['student 4 real name', 'student3_avatar_file_name'],
    'student_name_5': ['student 5 real name', 'student3_avatar_file_name'],
    'student_name_6': ['student 6 real name', 'student3_avatar_file_name'],
    }
```
I think the class name should be within the file (probably the first line), a JSON'd dictionary of the class data should be the second line.

* Refactor `create_classlist_data`
Function `create_classlist_data` in [class_functions](https://github.com/toonarmycaptain/dionysus/blob/master/dionysus_app/class_functions.py) will need to be refactored, which is nice, because it's a little ugly anyway.

Factor out the taking of class data into a function. 
Factor out the file write. 
* Tests
We'll need test asserting the dict JSON'd and written to disc is the same as that loaded and de-JSON'd. 
A test with dummy data asserting that the data in is the same as the data reloaded from disc would be nice too, although I've never worked out how to test a while loop well.

### PR DESCRIPTION
**Pull request by schedutron**
Fixes #25.

I've modified the `take_class_data_input()` function, but how do I access students' real names to be added to the class data?

### CODE REVIEW

**Comment by schedutron** [comment]
So there's no need to have a student's real name as one of the corresponding values? Just avatar name is enough?

**Comment by schedutron** [comment]
> Where have you changed it to store the class data as JSON rather than a plain comma and newline delimited string?

Didn't do that yet as I was a bit confused regarding the format. I've just changed the `take_class_data_input()` function so far, to return a dict instead of a comma-separated string.

**Comment by schedutron** [comment]
@toonarmycaptain Please review.

**Comment by toonarmycaptain** [comment]
Since it looks like we're gonna work through this in increments, I've created a branch, and moved this PR there. I haven't had time to look at it in detail yet, hopefully in the next 24 hrs.

**Comment by toonarmycaptain** [comment]
This looks fine so far.

File: dionysus_app/class_functions.py [hunk 1]
```diff
@@ -44,7 +44,8 @@ def create_classlist_data(classlist_name):  # TODO: fix path composition
         else:
             print(class_data)
 
-        classlist_file.write(class_data)  # consider using JSON?
+        # Can we use pickling instead of the following?
```
Comment (toonarmycaptain) [inline dionysus_app/class_functions.py]: What is the advantage in using pickle?

Comment (schedutron) [reply dionysus_app/class_functions.py]: Using pickle will allow us to restore data to Python data structures more directly. We can just dump a JSON in a pickle file and restore it later as a Python dictionary directly.

Comment (toonarmycaptain) [reply dionysus_app/class_functions.py]: If you use pickle, why bother using JSON? One can directly unpickle to a Python object. For the moment, while we're not saving anything more complex than a dictionary to disc, I prefer the readability of JSON.

**Comment by toonarmycaptain** [comment]
@all-contributors add @schedutron for code

**Pull request merged**